WalletWasabi/Wallets/P2pBlockProvider.cs

```diff
@@ -99,7 +99,12 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 					// Download block from selected node.
 					try
 					{
-						using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(RuntimeParams.Instance.NetworkNodeTimeout))) // 1/2 ADSL	512 kbit/s	00:00:32
+						// More permissive timeout if few nodes are connected to avoid exhaustion
+						var timeout = Nodes.ConnectedNodes.Count < 3
+							? Math.Min(RuntimeParams.Instance.NetworkNodeTimeout*1.5,600)
+							: RuntimeParams.Instance.NetworkNodeTimeout;
+
+						using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(timeout))) // 1/2 ADSL	512 kbit/s	00:00:32
```

Comment:
This comment doesnt have enough context and/or outdated... Would be great if this can be updated vis-a-vis this PR.

Reply:
The comment can be removed, I think it does not apply, nor give us any information - just confusion.

Reply:
agreed 👍 let's remove it @turbolay :)